src/pyclblast/src/pyclblast.pyx

```diff
@@ -2083,3 +2083,37 @@ def trsm(queue, m, n, a, b, a_ld, b_ld, alpha = 1.0, right_side = False, lower_t
     return cl.Event.from_int_ptr(<size_t>event)
 
 ####################################################################################################
+# Overrides the parameters
+####################################################################################################
+
+from libc.stdlib cimport malloc, free
+from libc.string cimport strdup
+
+cdef extern from "clblast_c.h":
+    ctypedef void* cl_device_id
+    CLBlastStatusCode CLBlastOverrideParameters(const cl_device_id device, const char* kernel_name, const CLBlastPrecision precision, const size_t num_parameters, const char** parameters_names, const size_t* parameters_values)
+
+def override_parameters(ctx, kernel_name, precision, parameters):
```

Comment:
My suggestion would be to keep the interface similar to the C++ interface, so replace `ctx` with `device` as argument, and then do something like this:
```
    cdef cl_device_id device_id = <cl_device_id><size_t>device.int_ptr
```
(see example above for OpenCL queues)

And then also add around line 104 of this file in a similar way as the queue is defined:
```
    struct _cl_device_id:
        pass
    ctypedef _cl_device_id* cl_device_id
```